nixos/lib/systemd-lib.nix

```
@@ -21,11 +21,12 @@ in rec {
         { preferLocalBuild = true;
           allowSubstitutes = false;
           inherit (unit) text;
+          passAsFile = [ "text" ];
         }
         ''
           name=${shellEscape name}
           mkdir -p "$out/$(dirname -- "$name")"
-          echo -n "$text" > "$out/$name"
+          cat "$textPath" > "$out/$name"
```

```suggestion
          mv "$textPath" "$out/$name"
```

`trivialBuilders.writeTextFile`[^1] uses `passAsFile = [ "text" ];` as well, but instead of `cat` redirecting or `cp`, it simply uses `mv`.
I believe this should be preferred here as well.

`nixosTests.systemd` pass with either solution though :)

[^1]: https://github.com/NixOS/nixpkgs/blob/280e17b0d73f65a9dc280d1c4c7c872d756ead70/pkgs/build-support/trivial-builders/default.nix#L159

---

Updated and tested, works well!